pkg/wallet/api.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package wallet
+
+// Wallet interface
+type Wallet interface {
+	Crypto
+	Pack
+}
+
+// Crypto interface
+type Crypto interface {
+
+	// CreateSigningKey create a new public/private signing keypair.
+	//
+	// Returns:
+	//
+	// KeyInfo: A `KeyInfo` representing the keypair
+	//
+	// error: error
+	CreateSigningKey(metadata map[string]string) (KeyInfo, error)
+
+	// GetSigningKey Fetch info for a signing keypair.
+	//
+	// Args:
+	//
+	// verkey: The verification key of the keypair
+	//
+	// Returns:
+	//
+	// KeyInfo: A `KeyInfo` representing the keypair
+	//
+	// error: error
+	GetSigningKey(verKey string) (KeyInfo, error)
+
+	// SignMessage sign a message using the private key associated with a given verification key.
+	//
+	// Args:
+	//
+	// message: The message to sign
+	//
+	// fromVerKey: Sign using the private key related to this verification key
+	//
+	// Returns:
+	//
+	// []byte: The signature
+	//
+	// error: error
+	SignMessage(message []byte, fromVerKey string) ([]byte, error)
+
+	// DecryptMessage decrypt message
+	//
+	// Args:
+	//
+	// encMessage: The encrypted message content
+	//
+	// toVerKey:The verification key of the recipient.
+	//
+	// []byte: Decrypted message content
+	//
+	// string: The sender verification key
+	//
+	// error: error
+	DecryptMessage(encMessage []byte, toVerKey string) ([]byte, string, error)
+}
+
+// Pack provide methods to pack and unpack msg
+type Pack interface {
+	// PackMessage Pack a message for one or more recipients.
+	//
+	// Args:
+	//
+	// message: The message to pack
+	//
+	// toVerKeys: The verification keys to pack the message for
+	//
+	// fromVerKey: The sender verification key
+	//
+	// Returns:
+	//
+	// []byte: The packed message
+	//
+	// error: error
+	PackMessage(message []byte, toVerKeys []string, fromVerKey string) ([]byte, error)
+
+	// UnpackMessage Unpack a message.
+	//
+	// Args:
+	//
+	// encMessage: The encrypted message
+	//
+	// Returns:
+	//
+	// string: message
+	//
+	// string: fromVerKey
+	//
+	// string: toVerKeys
+	//
+	// error: error
+	UnpackMessage(encMessage []byte) (string, string, string, error)
```

Comment:
Why doesn't `UnpackMessage` return `message []byte, toVerKeys []string, fromVerKey string, error` to match `PackMessage`?

Reply:
I also recommend returning a struct - there are too many return values.

Reply:
I will push the decrypt shortly... Crypter.Decrypt will give an idea of what will be returned.. not sure who in the middle will handle the exchange of keys needed. My thinking was the Pack()/UnPack() calls should already have access to the wallet keys.. to be discussed following my commit..

Reply:
done